### PR TITLE
Authentication Servlet

### PR DESCRIPTION
**Pull request by proyectoramirez**
Add servlets to handle authentication, which will be used in the future to track the authorship of comments.

There is no demo for this version, as there are no visual changes.

This PR may be merged independently.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/AuthServlet.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
+// Copyright 2019 Google LLC
```
Comment (Enriqe) [inline portfolio/src/main/java/com/google/sps/servlets/AuthServlet.java]: nit: 2020

File: portfolio/src/main/java/com/google/sps/servlets/AuthServlet.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    AuthStatus status;
+
+    if (userService.isUserLoggedIn()) {
+        String userEmail = userService.getCurrentUser().getEmail();
+
+        String urlToRedirectToAfterUserLogsOut = "/";
```
Comment (Enriqe) [inline portfolio/src/main/java/com/google/sps/servlets/AuthServlet.java]: Let's make this and `urlToRedirectToAfterUserLogsIn` file-level global constants.

https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names

Comment (proyectoramirez) [reply portfolio/src/main/java/com/google/sps/servlets/AuthServlet.java]: Done!

File: portfolio/src/main/java/com/google/sps/servlets/AuthServlet.java (hunk 1 above)
Comment (proyectoramirez) [reply portfolio/src/main/java/com/google/sps/servlets/AuthServlet.java]: Done

**Review by Enriqe — APPROVED** [review]
Nice!

**Pull request merged**